contracts/peripheral/WardQuestScheduler.sol

```diff
@@ -0,0 +1,206 @@
+    /** @notice Total number of epochs for the Quest  */
+    uint48 public constant duration = 2;
+    /** @notice Address of the ERC20 used for rewards  */
+    address public immutable cvx;
+    /** @notice Address  Ward Dark Quest Board */
+    address public immutable darkQuestBoard;
+
+    /** @notice List of Quest questId => stash */
+    mapping(uint256 => address) public quests;
+
+    /** @notice List of queued rewards (epoch => stash => amount) */
+    mapping(uint256 => mapping(address => uint256)) public rewardsQueue;
+
+    /** @notice Event emitted when rewards are queued  */
+    event QueuedRewards(address stash, uint256 epoch, uint256 rewardAmount);
+    /** @notice Event emitted when rewards are queued  */
+    event ForwardedRewards(address stash, uint256 epoch, uint256 rewardAmount);
+
+    /**
+     * @param _cvx  Cvx token contract
+     * @param _darkQuestBoard  Dark Quest Board
+     * @param _dao  Multisig address , the owner of the contract.
+     */
+    constructor(
+        address _cvx,
+        address _darkQuestBoard,
+        address _dao
+    ) Ownable() {
+        cvx = _cvx;
+        darkQuestBoard = _darkQuestBoard;
+        _transferOwnership(_dao);
+    }
+
+    /**
+     * @notice Creates a new Quest at a predefined duration (2 epochs).
+     * @dev Creates a new Quest struct, and QuestPeriods for the Quest duration
+     * @param stash Address of the Stash linked to this Quest.
+     * @param gauge Address of the Gauge targeted by the Quest
+     * @param objective Target bias to reach (equivalent to amount of veCRV in wei to reach)
+     * @param rewardPerVote Amount of reward per veCRV (in wei)
+     * @param totalRewardAmount Total amount of rewards for the whole Quest (in wei)
+     * @param feeAmount Paladin platform fees amount (in wei)
+     * @return questID : ID of the newly created Quest
+     */
+    function createQuest(
+        address stash,
+        address gauge,
+        uint256 objective,
+        uint256 rewardPerVote,
+        uint256 totalRewardAmount,
+        uint256 feeAmount,
+        address[] calldata blacklist
+    ) external returns (uint256 questID) {
+        require(stash != address(0), "!stash");
+        // Pull all the rewards + fee in this contract
+        IERC20(cvx).safeTransferFrom(msg.sender, address(this), totalRewardAmount.add(feeAmount));
+
+        questID = IDarkQuestBoard(darkQuestBoard).createQuest(
+            gauge,
+            cvx,
+            duration,
+            objective,
+            rewardPerVote,
+            totalRewardAmount,
+            feeAmount,
+            blacklist
+        );
+        quests[questID] = stash;
+    }
+
+    /**
+     * @notice Withdraw all undistributed rewards from Closed Quest Periods and queues them to it's linked stash.
+     * @param questID ID of the Quest
+     */
+    function withdrawAndQueueUnusedRewards(uint256 questID) external returns (uint256 amount) {
```

Comment:
Not need to protect with only owner but it could be possible. 
No need because only closed quest can be wihtdrawn , as well as the unused rewards are queued  on this contract, it never leaves the contract.